snapcraft/commands/remote.py

```diff
@@ -143,8 +149,56 @@ def _validate(self, parsed_args: argparse.Namespace) -> None:
                     "In non-interactive runs, please use the option "
                     "`--launchpad-accept-public-upload`."
                 ),
+                doc_slug="/explanation/remote-build.html",
                 reportable=False,
-                retcode=77,
+                retcode=os.EX_NOPERM,
+            )
+
+        build_for = parsed_args.remote_build_build_for or None
+        platform = parsed_args.remote_build_platform or None
+        parameter = "--build-for" if build_for else "--platform" if platform else None
+        keyword = (
+            "architectures"
+            if self.project._architectures_in_yaml
+            else "platforms" if self.project.platforms else None
+        )
+
+        if keyword and parameter:
+            raise errors.RemoteBuildError(
+                f"{parameter!r} cannot be used when {keyword!r} is in the snapcraft.yaml.",
+                resolution=f"Remove {parameter!r} from the command line or remove {keyword!r} in the snapcraft.yaml.",
+                doc_slug="/explanation/remote-build.html",
+                retcode=os.EX_CONFIG,
+            )
+
+        if platform:
+            if self.project.get_effective_base() == "core22":
+                raise errors.RemoteBuildError(
+                    "'--platform' cannot be used for core22 snaps.",
+                    resolution="Use '--build-for' instead.",
+                    doc_slug="/explanation/remote-build.html",
+                    retcode=os.EX_CONFIG,
+                )
+            if platform not in SUPPORTED_ARCHS:
```

Comment:
This is counterintuitive to me. It's correct behaviour for now, but I'd love to see if we could get `--platform` to behave the same way it does for `pack`

Reply:
I totally agree and I have an issue for it here: https://github.com/canonical/snapcraft/issues/4992